specifyweb/frontend/js_src/lib/components/Preferences/UserDefinitions.tsx

```diff
@@ -2058,15 +2082,79 @@ import('../DataModel/tables')
           ),
           'Unable to find tree full name value'
         );
-        overwriteReadOnly(name, 'title', getField(tables.Taxon, 'name').label);
+        overwriteReadOnly(
+          name, 
+          'title', 
+          getField(tables.Taxon, 'name').label);
         overwriteReadOnly(
           fullName,
           'title',
           getField(tables.Taxon, 'fullName').label
         );
       } else softError('Unable to replace the tree preferences item title');
+
+      // TODO: Refactor this with a helper function since this is largely the same as above
+
+      // Update titles for orderByField
+      const treeOrderByBehavior =
+        userPreferenceDefinitions.treeEditor.subCategories.behavior.items.orderByField;
+      if ('values' in treeOrderByBehavior) {
+        const orderByValues = treeOrderByBehavior.values as RA<{
+          readonly value: string;
+          readonly title: string;
+        }>;
+
+        const nameOrderBy = defined(
+          orderByValues.find(
+            (entry) => typeof entry === 'object' && entry.value === 'name'
+          ),
+          'Unable to find tree name value for orderByField'
+        );
+        const fullNameOrderBy = defined(
+          orderByValues.find(
+            (entry) => typeof entry === 'object' && entry.value === 'fullName'
+          ),
+          'Unable to find tree full name value for orderByField'
+        );
+        const rankId = defined(
+          orderByValues.find(
+            (entry) => typeof entry === 'object' && entry.value === 'rankId'
+          ),
+          'Unable to find tree rankId value'
+        );
+        const nodeNumber = defined(
+          orderByValues.find(
+            (entry) => typeof entry === 'object' && entry.value === 'nodeNumber'
+          ),
+          'Unable to find tree nodeNumber value'
+        );
+
+        overwriteReadOnly(
```

Comment:
all these overrides of .title were necessary earlier because schema localization is not yet loaded when preferences are defined

however, we since added ability to define title as a function rather than a static string:
![image](https://github.com/user-attachments/assets/9fc03f3e-f235-476e-9ec4-9110481e0db6)

we already have a tableLabel function -which gets the label if table is already fetched, or if not, does gracefully fallback (which should generally never be visible to the user)

![image](https://github.com/user-attachments/assets/93483d37-8b20-42cb-8f0e-8bc937ef63c5)

if you add a similar small utility called `fieldLabel`, then that would remove need for all these `overwriteReadOnly` calls, and reduce quite a bit of code

Reply:
Partially resolved in [`b80794a` (#6046)](https://github.com/specify/specify7/pull/6046/commits/b80794a9cfb59ee7b72ec9d58fdb4ac8bcfb6943)

Reply:
this is how `fieldLabel()` function could look like:

```ts
const fieldLabel = <TABLE_NAME extends keyof Tables>(tableName: TABLE_NAME, fieldName: TableFields<Tables[TABLE_NAME]>): LocalizedString =>
  genericTables[tableName].field[fieldName]?.label ?? camelToHuman(fieldName);
```

but I can see that picklist-like preferences don't yet support function as `title`, so using the above function requires a bit more code changes - it's fine to leave things as is though